src/module.ts

```diff
@@ -32,7 +32,7 @@ export default defineNuxtModule<Partial<Options>>({
         transformPlugin.vite({
           include: options.include,
           exclude: options.exclude,
-          sourcemap: nuxt.options.sourcemap[mode],
+          sourcemap: nuxt.options.sourcemap[mode] as boolean,
```

Comment:
Using `as` is not the best idea, perhaps we should optimize the sourcemap type in the `PluginOptions`.